### PR TITLE
Sped up the relaunch of the share button.

### PR DESCRIPTION
**Pull request by petermisik**

### CODE REVIEW

File: Logging/Controllers/View Controllers/Form 781/Form781ViewController.swift [hunk 1]
```diff
@@ -84,8 +84,10 @@ class Form781ViewController: UIViewController {
         // Load the PDF
         
         NSLog("***** Send button tapped *****")
-        let _ = Helper.exportPDF()
-        NSLog("***** PDF Saved to disc *****")
+        if !FileManager.default.fileExists(atPath: Helper.convertURLtoString(fileURL: Helper.getDocDir().appendingPathComponent(Helper.pdfTitle))){
+            let _ = Helper.exportPDF()
+            NSLog("***** PDF Saved to disc *****")
```
Comment (Pete-Hoch) [inline Logging/Controllers/View Controllers/Form 781/Form781ViewController.swift:89]: Is the PDF file deleted when the data changes? If not, we will use an outdated image.

**Review by Pete-Hoch — CHANGES_REQUESTED** [review]

File: Logging/Controllers/View Controllers/Form 781/Form781ViewController.swift (hunk 1 above)
Comment (petermisik) [reply Logging/Controllers/View Controllers/Form 781/Form781ViewController.swift:96]: It is not.  Do we have an observer in place to catch a data change?

**Comment by Pete-Hoch** [comment]
Also, I crashed here:

#0	0x00007fff60c58462 in __pthread_kill ()
#1	0x00007fff60c91610 in pthread_kill ()
#2	0x00007fff200fab94 in abort ()
#3	0x00007fff2fbffc05 in swift::fatalError(unsigned int, char const*, ...) ()
#4	0x00007fff2fbf8097 in swift::swift_dynamicCastFailure(void const*, char const*, void const*, char const*, char const*) ()
#5	0x00007fff2fbf810a in swift::swift_dynamicCastFailure(swift::TargetMetadata<swift::InProcess> const*, swift::TargetMetadata<swift::InProcess> const*, char const*) ()
#6	0x00007fff2fbf9cec in swift_dynamicCastImpl(swift::OpaqueValue*, swift::OpaqueValue*, swift::TargetMetadata<swift::InProcess> const*, swift::TargetMetadata<swift::InProcess> const*, swift::DynamicCastFlags) ()
#7	0x000000010b72338d in Numeric.drawIn(_:) at /Users/hoch/Dev/Logging-iOS/Logging/HelperFiles/ImageGenerator.swift:204
#8	0x000000010b71c96a in closure #1 in static ImageGenerator.generateFilledFormPageOneImage(from:) at /Users/hoch/Dev/Logging-iOS/Logging/HelperFiles/ImageGenerator.swift:33
#9	0x000000010b7234ac in thunk for @callee_guaranteed (@guaranteed UIGraphicsImageRendererContext) -> () ()
#10	0x000000010b7234fc in thunk for @escaping @callee_guaranteed (@guaranteed UIGraphicsImageRendererContext) -> () ()
#11	0x00007fff242332ee in -[UIGraphicsRenderer runDrawingActions:completionActions:format:error:] ()
#12	0x00007fff2423313e in -[UIGraphicsRenderer runDrawingActions:completionActions:error:] ()
#13	0x00007fff242309f0 in -[UIGraphicsImageRenderer imageWithActions:] ()
#14	0x000000010b71bd97 in static ImageGenerator.generateFilledFormPageOneImage(from:) at /Users/hoch/Dev/Logging-iOS/Logging/HelperFiles/ImageGenerator.swift:21
#15	0x000000010b754ade in static Helper.generateSideOfForm(side:) at /Users/hoch/Dev/Logging-iOS/Logging/HelperFiles/HelperFunctions.swift:331
#16	0x000000010b756721 in static Helper.exportPDF() at /Users/hoch/Dev/Logging-iOS/Logging/HelperFiles/HelperFunctions.swift:372
#17	0x000000010b73230b in Form781ViewController.sendButtonTapped(_:) at /Users/hoch/Dev/Logging-iOS/Logging/Controllers/View Controllers/Form 781/Form781ViewController.swift:88
#18	0x000000010b732fa4 in @objc Form781ViewController.sendButtonTapped(_:) ()
#19	0x00007fff246972b2 in -[UIApplication sendAction:to:from:forEvent:] ()
#20	0x00007fff23fc08a0 in -[UIControl sendAction:to:forEvent:] ()
#21	0x00007fff23fc0bc3 in -[UIControl _sendActionsForEvents:withEvent:] ()
#22	0x00007fff23fbf4ac in -[UIControl touchesEnded:withEvent:] ()
#23	0x00007fff246d3962 in -[UIWindow _sendTouchesForEvent:] ()
#24	0x00007fff246d57db in -[UIWindow sendEvent:] ()
#25	0x00007fff246af57a in -[UIApplication sendEvent:] ()
#26	0x00007fff24740000 in __processEventQueue ()
#27	0x00007fff24736973 in __eventFetcherSourceCallback ()
#28	0x00007fff2038f38a in __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ ()
#29	0x00007fff2038f282 in __CFRunLoopDoSource0 ()
#30	0x00007fff2038e764 in __CFRunLoopDoSources0 ()
#31	0x00007fff20388f2f in __CFRunLoopRun ()
#32	0x00007fff203886d6 in CFRunLoopRunSpecific ()
#33	0x00007fff2bededb3 in GSEventRunModal ()
#34	0x00007fff24690e0b in -[UIApplication _run] ()
#35	0x00007fff24695cbc in UIApplicationMain ()

I think this bug snuck in on us. Looks like we need a unit test for generating the PDF.

        let s = NSAttributedString(string: self as! String , attributes: standardAttributes)

self is a Numeric. This may work:

        let s = NSAttributedString(string: "\(self)", attributes: standardAttributes)

But, tapping on "SEND ->" several times in a row yields this logging.

2020-12-16 13:15:45.898500-1000 Logging[15343:1376768] ***** Send button tapped *****
2020-12-16 13:15:48.273355-1000 Logging[15343:1376768] ***** PDF Saved to disc *****
2020-12-16 13:15:48.274060-1000 Logging[15343:1376768] ***** PDF Loaded *****
2020-12-16 13:15:54.146966-1000 Logging[15343:1377045] [ShareSheet] connection invalidated
2020-12-16 13:15:54.841914-1000 Logging[15343:1376768] ***** Send button tapped *****
2020-12-16 13:15:57.170492-1000 Logging[15343:1376768] ***** PDF Saved to disc *****
2020-12-16 13:15:57.171120-1000 Logging[15343:1376768] ***** PDF Loaded *****
2020-12-16 13:16:00.446600-1000 Logging[15343:1377154] [ShareSheet] connection invalidated
2020-12-16 13:16:02.929891-1000 Logging[15343:1376768] ***** Send button tapped *****
2020-12-16 13:16:05.199465-1000 Logging[15343:1376768] ***** PDF Saved to disc *****
2020-12-16 13:16:05.200038-1000 Logging[15343:1376768] ***** PDF Loaded *****
2020-12-16 13:16:07.465896-1000 Logging[15343:1377394] [ShareSheet] connection invalidated
2020-12-16 13:16:09.681203-1000 Logging[15343:1376768] ***** Send button tapped *****
2020-12-16 13:16:11.968496-1000 Logging[15343:1376768] ***** PDF Saved to disc *****
2020-12-16 13:16:11.969232-1000 Logging[15343:1376768] ***** PDF Loaded *****
2020-12-16 13:16:15.193374-1000 Logging[15343:1377448] [ShareSheet] connection invalidated

So we are not saving any time.

**Comment by jbethanco** [comment]
Sorry about the numeric crash. Thought it passed my tests here. Will do the interpolation fix.

**Pull request closed without merging**